vaadin-portlet/src/main/java/com/vaadin/flow/portal/VaadinPortletResponse.java

```diff
@@ -127,17 +128,7 @@ public void setDateHeader(String name, long timestamp) {
 
     @Override
     public void setCacheTime(long milliseconds) {
-        // FIXME This is copy paste
-        if (milliseconds <= 0) {
-            setHeader("Cache-Control", "no-cache");
-            setHeader("Pragma", "no-cache");
-            setDateHeader("Expires", 0);
-        } else {
-            setHeader("Cache-Control", "max-age=" + milliseconds / 1000);
-            setDateHeader("Expires", System.currentTimeMillis() + milliseconds);
-            // Required to apply caching in some Tomcats
-            setHeader("Pragma", "cache");
-        }
+        VaadinServletResponse.doSetCacheTime(this, milliseconds);
```

Comment:
`doSetCacheTime` is private in VSR

Reply:
Oh. That has been changed since Vaadin 8. There it is package protected and VaadinPortletResponse is in the same package.

https://github.com/vaadin/framework/blob/master/server/src/main/java/com/vaadin/server/VaadinServletResponse.java#L65

But as in Flow Portlet is a different project, the same approach would require making the method in VSR public.

Reply:
Unfortunately yes, or in a more advanced way.